frontend/app/module/election/page/ElectionReportPage.tsx

```diff
@@ -31,22 +50,29 @@ export function ElectionReportPage() {
       })
       .then(
         (blob) => {
-          const file = new File([blob], filename);
-          const fileUrl = window.URL.createObjectURL(file);
-          const anchorElement = document.createElement("a");
-
-          anchorElement.href = fileUrl;
-          anchorElement.download = filename;
-          anchorElement.hidden = true;
-
-          document.body.appendChild(anchorElement);
+          offerDownload(blob, filename);
+        },
+        (error: unknown) => {
+          console.error(error);
+        },
+      );
+  }
 
-          anchorElement.click();
-          anchorElement.remove();
+  function downloadZipResults() {
+    let filename: string;
+    fetch(`/api/elections/${election.id}/download_zip_results`)
+      .then((result) => {
+        if (result.status !== 200) {
+          const message = `Failed to download zip: status code ${result.status}`;
 
-          setTimeout(() => {
-            window.URL.revokeObjectURL(fileUrl);
-          }, 30000);
+          throw new Error(message);
+        }
+        filename = result.headers.get("Content-Disposition")?.split('filename="')[1]?.slice(0, -1) ?? "document";
+        return result.blob();
+      })
+      .then(
+        (blob) => {
+          offerDownload(blob, filename);
         },
```

Comment:
I think we should try to completely deduplicate the logic that is shared with the `downloadPdfResults` function.

We can make these async functions (and annotate the `onClick` function for the button as `void`) to simplify the function.